### PR TITLE
DateBox: add Vue and React code snippets

### PR DESCRIPTION
**Pull request by ksercs**

### CODE REVIEW

File: concepts/05 Widgets/DateBox/05 Control the Behavior.md [hunk 1]
```diff
@@ -37,6 +37,62 @@ The **DateBox** widget allows an end user to pick or type a value. To specify wh
...
+            this.date = new Date();
+        }
+
+        render() {
+            return (
+                <DateBox
+                    defaultValue={this.date}
+                    type="date"
+                    pickerType="calendar"
+                    applyValueMode="useButtons" // or "instantly"
```
Comment (AlexJameson) [inline concepts/05 Widgets/DateBox/05 Control the Behavior.md]:
```suggestion
                    applyValueMode="useButtons" {/* or "instantly" */}
```

**Review by AlexJameson — APPROVED** [review]

File: concepts/05 Widgets/DateBox/05 Control the Behavior.md (hunk 1 above)
Comment (ksercs) [reply concepts/05 Widgets/DateBox/05 Control the Behavior.md]: fixed

File: concepts/05 Widgets/DateBox/04 Value Formatting and Masked Input.md [hunk 2]
```diff
@@ -39,6 +39,49 @@ In addition to value formatting, the **displayFormat** can be used as a mask to
...
+    import 'devextreme/dist/css/dx.light.css';
+
+    import DateBox from 'devextreme-react/date-box';
+
+    class App extends React.Component {
+        render() {
+            return (
+                <DateBox
+                    useMaskBehavior={true}
+                    displayFormat="EEEE, d 'of' MMM" // "Tuesday, 16 of Oct"
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/DateBox/04 Value Formatting and Masked Input.md]:
```suggestion
                    displayFormat="EEEE, d 'of' MMM" {/* "Tuesday, 16 of Oct" */}
```

File: concepts/05 Widgets/DateBox/10 Handle the Value Change Event.md [hunk 3]
```diff
@@ -35,6 +35,63 @@ If instead of picking a value, an end user types it into the **DateBox**, this v
...
+            return (
+                <DateBox
+                    defaultValue={this.date}
+                    type="date"
+                    valueChangeEvent="keyup"
+                />
+            );
+        }
+    }
+    export default App;
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/DateBox/10 Handle the Value Change Event.md:93]: What about the `onValueChanged` example from below? Let's add it and also show two-way binding to the `value` option in it, like we did [for the Calendar](https://github.com/DevExpress/devextreme-documentation/pull/353/files#diff-2e1e9256369f8552a879e3d5bdd78278).

**Review by RomanTsukanov — CHANGES_REQUESTED** [review]

File: concepts/05 Widgets/DateBox/04 Value Formatting and Masked Input.md (hunk 2 above)
Comment (ksercs) [reply concepts/05 Widgets/DateBox/04 Value Formatting and Masked Input.md]: fixed

File: concepts/05 Widgets/DateBox/10 Handle the Value Change Event.md (hunk 3 above)
Comment (ksercs) [reply concepts/05 Widgets/DateBox/10 Handle the Value Change Event.md:93]: fixed

File: concepts/05 Widgets/DateBox/10 Handle the Value Change Event.md [hunk 4]
```diff
@@ -58,16 +115,16 @@ To process a new **DateBox** value, you need to handle the value change event. I
 
     <!--HTML-->
     <dx-date-box ...
-        (onValueChanged)="dateBox_valueChanged($event)">
+        [(onValueChanged)]="handleValueChange($event)">
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/DateBox/10 Handle the Value Change Event.md]: I meant two-way binding to the `value`.

File: concepts/05 Widgets/DateBox/10 Handle the Value Change Event.md [hunk 5]
```diff
@@ -79,6 +136,69 @@ To process a new **DateBox** value, you need to handle the value change event. I
...
+
+    import DxDateBox from 'devextreme-vue/date-box';
+
+    export default {
+        components: {
+            DxDateBox
+        },
+        data() {
+            return {
+                handleValueChange: function (e) {
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/DateBox/10 Handle the Value Change Event.md]: Functions should be declared in the `methods` object.

File: concepts/05 Widgets/DateBox/10 Handle the Value Change Event.md [hunk 6]
```diff
@@ -79,6 +136,69 @@ To process a new **DateBox** value, you need to handle the value change event. I
...
+
+        handleValueChange(e) {
+            const previousValue = e.previousValue;
+            const newValue = e.value;
+            // Event handling commands go here
+        }
+
+        render() {
+            return (
+                <DateBox
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/DateBox/10 Handle the Value Change Event.md:207]: Let's add the state and the value.

**Review by RomanTsukanov — CHANGES_REQUESTED** [review]

File: concepts/05 Widgets/DateBox/10 Handle the Value Change Event.md [hunk 7]
```diff
@@ -79,6 +138,82 @@ To process a new **DateBox** value, you need to handle the value change event. I
...
+                date: newValue
+            });
+
+            // Event handling commands go here
+        }
+
+        render() {
+            return (
+                <DateBox
+                    value={this.state.date},
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/DateBox/10 Handle the Value Change Event.md]:
```suggestion
                    value={this.state.date}
```

File: concepts/05 Widgets/DateBox/10 Handle the Value Change Event.md [hunk 8]
```diff
@@ -79,6 +138,82 @@ To process a new **DateBox** value, you need to handle the value change event. I
...
+
+        handleValueChange(e) {
+            const previousValue = e.previousValue;
+            const newValue = e.value;
+
+            this.setState({
+                date: newValue
+            });
+
+            // Event handling commands go here
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/DateBox/10 Handle the Value Change Event.md]:
```suggestion

```

File: concepts/05 Widgets/DateBox/10 Handle the Value Change Event.md [hunk 9]
```diff
@@ -79,6 +138,82 @@ To process a new **DateBox** value, you need to handle the value change event. I
...
+        }
+
+        handleValueChange(e) {
+            const previousValue = e.previousValue;
+            const newValue = e.value;
+
+            this.setState({
+                date: newValue
+            });
+
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/DateBox/10 Handle the Value Change Event.md]:
```suggestion

```

File: concepts/05 Widgets/DateBox/10 Handle the Value Change Event.md [hunk 10]
```diff
@@ -79,6 +138,82 @@ To process a new **DateBox** value, you need to handle the value change event. I
...
+            this.state = {
+                date: new Date()
+            };
+            this.handleValueChange = this.handleValueChange.bind(this);
+        }
+
+        handleValueChange(e) {
+            const previousValue = e.previousValue;
+            const newValue = e.value;
+
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/DateBox/10 Handle the Value Change Event.md:199]:
```suggestion
            // Event handling commands go here
```

**Review by RomanTsukanov — CHANGES_REQUESTED** [review]

File: concepts/05 Widgets/DateBox/10 Handle the Value Change Event.md [hunk 11]
```diff
@@ -79,6 +138,81 @@ To process a new **DateBox** value, you need to handle the value change event. I
...
+    import DxDateBox from 'devextreme-vue/date-box';
+
+    export default {
+        components: {
+            DxDateBox
+        },
+        data() {
+            return {
+                date: new Date(),
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/DateBox/10 Handle the Value Change Event.md]:
```suggestion
                date: new Date()
```

**Review by RomanTsukanov — CHANGES_REQUESTED** [review]

**Review by RomanTsukanov — APPROVED** [review]

**Pull request merged**